File: Python/validParentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        if s[0] == ")" or s[0] == "}" or s[0] == "]":
            return False
        res = []
        for i in s:
            if i == "(" or i == "{" or i == "[":
                res.append(i)
            elif len(res) == 0:
                return False
            elif i == ")" and res[len(res)-1] != "(":
                return False
            elif i == "}" and res[len(res)-1] != "{":
                return False
            elif i == "]" and res[len(res)-1] != "[":
                return False
            else:
                res.pop()
        
        if len(res) != 0:
            return False
        else:
            return True

File: JavaScript/validParentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    let res = [];
    for (let i of s) {
        if(i === "(" || i === "{" || i === "[") {
            res.push(i);
        } else if (res.length === 0) {
            return false;
        } else if (i === ")" && res[res.length-1] !== "(") {
            return false;
        } else if (i === "}" && res[res.length-1] !== "{") {
            return false;
        } else if (i === "]" && res[res.length-1] !== "[") {
            return false;
        } else {
            res.pop();
        }
    }
    if (res.length !== 0) {
        return false;
    }
    return true;
};